API/models/schemas/refs/types.json

```diff
@@ -21,6 +21,18 @@
         },
         "slug": {
             "pattern": "^[a-z0-9-_]+$"
+        },
+        "name": {
+            "pattern": "^\\w(\\w|\\-)*$"
+        },
+        "id": {
+            "pattern": "^\\w(\\w|\\-)*(\\.\\w(\\w|\\-)*)*$"
+        },
+        "idWithWildcards": {
+            "pattern": "^(\\w|\\*)(\\w|\\-|\\*)*(\\.(\\w|\\*)(\\w|\\-|\\*)*)*$"
```

Comment:
*******.*******.****something**** is going to be a match

Reply:
yes, but it is not a problem, any amount of * is possible for wildcards. In the applicability what is used is: if it is exactly ** (for example /Physical/BASIC/**) then it is recursive applicability, otherwise the * is replaced by any character at least once, so for example /Physical/BASIC/ab if it meets /Physical/BASIC/a* but not /Physical/BASIC/a**

Reply:
So /Physical/siteA/* applies to everyone directly under siteA, /Physical/siteA/** to everyone under it in any level, /Physical/siteA/*** to everyone directly under it that has a name with 3 chars ?

Reply:
I was thinking this through and I'm going to change the behavior to be exactly the same as in linux (https://en.m.wikipedia.org/wiki/Glob_(programming)#Unix-like), so:
\* becomes 0 or more characters, so ls a* and ls a\*\*\* are the same, no need to limit the number of stars.
\*\* in general would also be the same as * (ls a* and ls a\*\* give the same result), but in parcular if used as a directory it generates the recursion (ls \*\*/a).

Maybe that ** is a particular case is not the best, but I think that the clearest thing to do is to follow the standard of how the linux console works. Do you agree?

Reply:
But the wildcard would still apply to only one hierarchy level except for the ** then? I think this one and the wildcard implementation of /api/objects (used by the cli with a get site*, for example) should be as similar as possible. 
For that endpoint, the current implementation is something like:
/api/objects?id=siteA.* => returns all direct children of siteA (buildings)
/api/objects?id=siteA.*.* => returns all grandchildren of siteA (rooms)
/api/objects?id=*.*.R1 => returns all rooms called R1 (different parents possible)
Each wildcard usage applies to only one hierarchy level. We only thought of using one * as useful here, the use cases we wanted to handle did not need more than that. However, the user is not forbidden to use multiple * (although no one expressed the need for it yet) and the response would be like: 
/api/objects?id=siteA.** => returns all direct children of siteA with at least two chars
/api/objects?id=siteA.t* => returns all direct children of siteA that begin with t and followed by at least one chars
/api/objects?id=siteA.t*** => returns all direct children of siteA that begin with t and followed by at least 3 chars

Reply:
I did the change in 7d146065. I am now using a library that implements the linux standard. Yes, as you said "*" is only one hierarchy level no matter how many, with the exception of /**/ (.**. at the id level) which is what adds the multiple hierarchy. Using this library also allows us to add other wildcards that I added to the documentation. At the moment, these changes don't affect the api endpoints at all, they are only for the applicability of the layers. In the next pull requests, when the filtering ls is added, the behaviour of these endpoints will be modified to support multiple hierarchy levels.